[OttoTan] iP

OttoTan opened the pull request:
### Megamind

> "Time is money" - Benjamin Franklin

Your one stop solution for planners. It is,

- User friendly
- Intuitive commands
- Lightweight

It is super easy to use

1. download it from [here](https://github.com/nus-cs2103-AY2425S1/ip/pull/98)
2. open it
3. add your tasks
4. enjoy your increase productivity 💯💯

The program is **free** and has tons of features!

- [ ] Adding tasks with deadlines and time periods
- [ ] Searching for tasks by keywords
- [ ] Deleting tasks after you are done

If you want to add your own features, you can take a look at the `Megamind` class:

```Java
public class Megamind {

// Add your functions here

}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Added functions in Megamind class for skeletal version

Commit:
Made logo more readable

Commit:
Implemented simple echoing of user message
Implemented exit of chatbot upon "bye"

Commit:
Implemented simple echoing of user message
Implemented exit of chatbot upon "bye"

Commit:
Merge tag 'Level-1'

Commit:
Implemented tracking, adding and displaying of tasks in list

Commit:
Implemented the marking and unmarking of task
Introduced a Task class to more naturally represent the tasks

Commit:
Introduced todo, deadline and event classes that extends the task class to naturally represent them
Implemented the logic within the chatbot

Commit:
Implemented TextUiTesting

Commit:
Implemented error handling within the main loop as well as other functions

Commit:
Implemented deletion of tasks

Commit:
Update automated tests

Commit:
Implement local storage of tasks. Saving and loading previous file data is now allowed. Data is saved in a .ser file, using Java's serialisation rather than a .txt to prevent additional parsing of data and changing of current code base

OttoTan committed:
Merge pull request #1 from OttoTan/branch-Level-7

Implement local storage of tasks. Saving and loading previous file da…

Commit:
Implement storage of time strings as Java LocalDateTime objects

OttoTan committed:
Merge pull request #2 from OttoTan/branch-Level-8

Implement storage of time strings as Java LocalDateTime objects

Commit:
Introduce Parser, Storage and Ui classes to improve the codes OOP principles

OttoTan committed:
Merge pull request #3 from OttoTan/branch-A-MoreOOP

Introduce Parser, Storage and Ui classes to improve the codes OOP principles

Commit:
Organize the classes into suitable java packages

OttoTan committed:
Merge pull request #4 from OttoTan/branch-A-Packages

Organize the classes into suitable java packages

Commit:
Merge branch 'add-gradle-support'

Commit:
Add JUnit tests to test the behavior of the code

OttoTan committed:
Merge pull request #5 from OttoTan/branch-A-JUnit

Add JUnit tests to test the behavior of the code

Commit:
Package the app as an executable JAR file so that it can be distributed easily

OttoTan committed:
Merge pull request #6 from OttoTan/branch-A-Jar

Package the app as an executable JAR file so that it can be distributed easily

Commit:
Give users a way to find a task by searching for a keyword

OttoTan committed:
Merge pull request #7 from OttoTan/branch-Level-9

Give users a way to find a task by searching for a keyword

Commit:
Update and include JavaDocs for Parser functions

Commit:
Update and include JavaDocs for Parser functions

OttoTan committed:
Merge branch 'master' into branch-A-JavaDoc

OttoTan committed:
Merge pull request #8 from OttoTan/branch-A-JavaDoc

Update and include JavaDocs for Parser functions

Commit:
Reformat all files in main folder

This is to upkeep standardization and improve readability of codebase

OttoTan committed:
Merge pull request #9 from OttoTan/branch-A-CodingStandard

Reformat all files in main folder

Commit:
Checkstyle all files in the main and test folder

This is to standardize all code within the code base

OttoTan committed:
Merge pull request #10 from OttoTan/branch-A-CodingStandard

Checkstyle all files in the main and test folder

OttoTan committed:
Merge pull request #11 from OttoTan/master

Update JavaDoc branch

Commit:
Add Javadoc comments for class header

When Gradle checkstyle was run, it was noted that there were missing JavaDoc comments for classes, rather than functions

With this change, it will improve readability of the code base

OttoTan committed:
Merge pull request #12 from OttoTan/branch-A-JavaDoc

Add Javadoc comments for class header

Commit:
Will implement front end GUI for the chatbot

Currently, user interface is through CLI commands, which may not be intuitive for users

Adding this GUI will improve user-friendliness

OttoTan committed:
Merge pull request #13 from OttoTan/branch-Level-10

Will implement front end GUI for the chatbot

Commit:
Add JavaDocs for GUI classes

Ensured all methods and constructors are documented.

OttoTan committed:
Merge pull request #14 from OttoTan/branch-A-CheckStyle

Add JavaDocs for GUI classes

WuJinhan1 commented on src/main/java/megamind/gui/DialogBox.java:
```diff
@@ -0,0 +1,82 @@
+ * user and cookie dialog boxes.
+ */
+public class DialogBox extends HBox {
+    @FXML
```
LGTM!

WuJinhan1 commented on src/main/java/megamind/exception/InvalidCommandException.java:
```diff
@@ -0,0 +1,16 @@
+ * Exception thrown when an invalid command is encountered.
+ */
+public class InvalidCommandException extends Exception {
+
```
I like how you separated all the exceptions into different ones specific to the exception it pertains to.

WuJinhan1 commented on src/main/java/megamind/gui/Main.java:
```diff
@@ -0,0 +1,49 @@
+ * the main window of the application.
+ */
+public class Main extends Application {
+    private final Megamind megamind = new Megamind();
```
LGTM!

WuJinhan1 commented on src/main/java/megamind/main/Launcher.java:
```diff
@@ -0,0 +1,13 @@
+/**
+ * Acts as the entry point for Cookie application
+ */
+public class Launcher {
```
I like how you split up the different files into different packages so that it is easier to read!

WuJinhan1 commented on src/main/resources/images/Megamind.png:
I like the use of Megamind as the face of the chatbot!

WuJinhan1 commented on src/main/java/megamind/task/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+ * It extends the `Task` class and includes additional information about the deadline of the task.
+ * The class provides methods to construct a deadline task and return its string representation.
+ */
+public class Deadline extends Task {
```
I like how you separated the tasks into different files instead of clumping it all into one, makes the code cleaner.

WuJinhan1 commented on src/main/java/megamind/ui/Ui.java:
```diff
@@ -0,0 +1,79 @@
+     * Greets the user.
+     */
+    public String greet() {
+        String logo = """
```
I like how you changed the logo of the original chatbot to make it look more like megamind!

WuJinhan1 reviewed (commented):
Overall, LGTM! I like how you implemented a character behind the chatbot so that it has more character.

Commit:
Add assertions to document important assumptions

Important assumptions were not tested beforehand, which may have led to undesirable side effects

- Added assertions in `MainWindow` class to ensure `megamind` and `input` are not null
- Added assertions in `Main` class to ensure `fxmlLoader` successfully loads the FXML file and `megamind` is not null
- Added assertions in `DialogBox` class to ensure `text` and `img` are not null, and `tmp` is not empty before reversing the list
- Added assertions in `Megamind` class to ensure `command` is not null or empty, and `index` is within bounds
- Added assertions in `Storage` class to ensure `tasks` is not null and the file exists before loading tasks
- Corrected type assertion in `Megamind` constructor to ensure `taskList.getTasks()` is an instance of `ArrayList<Task>`

OttoTan committed:
Merge pull request #15 from OttoTan/branch-A-Assertions

Add assertions to document important assumptions

Commit:
Refactor and improve code quality in `Parser`, `Ui`, `Megamind` and `Storage` classes

Code although succinct, could have improved in readability

- Extracted validation logic in `Parser` class to separate methods for better readability and maintainability
- Simplified parsing methods in `Parser` class by breaking down complex methods into smaller, more focused methods
- Improved exception handling in `Parser` class to provide more specific error messages
- Ensured proper directory creation and file existence checks in `Storage` class
- Added detailed comments to explain the purpose and functionality of methods in both `Parser` and `Storage` classes
- Enhanced code readability and maintainability by following best practices and coding standards

OttoTan committed:
Merge pull request #16 from OttoTan/branch-A-CodeQuality

Refactor and improve code quality in `Parser`, `Ui`, `Megamind` and `…

Commit:
Implement Continuous Integration using GitHub actions

Local repositories had to manually pull from origin whenever the master branch was updated

By implementing this, this process is automated

Commit:
Amend minor checkstyle errors

Commit:
Updated script files

Commit:
Remove unnecessary I/O redirection tests as they are not applicable to this project

Commit:
Fix a couple of typos in the test cases

Commit:
Fix a couple of typos in the test cases

OttoTan committed:
Merge pull request #18 from OttoTan/branch-A-CI

Implement CI using GitHub Actions

Commit:
Implement a new Recurring Task, where users can set their start, end and intervals between tasks

Users have to consistently manually add tasks in if they want a constant task manager

By implementing this feature, it will improve user experience

OttoTan committed:
Merge pull request #19 from OttoTan/branch-B-RecurringTasks

Implement a new Recurring Task, where users can set their start, end …

Commit:
Fix checkstyle errors

OttoTan committed:
Merge pull request #20 from OttoTan/branch-A-CheckStyle

Fix checkstyle errors

Commit:
Correct some of the JavaDocs that were affected by the latest refactoring of code

Commit:
Add user instructions and features to the README and include a representative screenshot

User guide was barebones, which may cause users confusion when trying it out of the first time

Commit:
Update README.md with user guide for Megamind

- Added sections for deleting tasks, and exiting Megamind.
- Included examples and formats for each command.

Commit:
Fix minor checkstyle errors in Megamind class JavaDocs

Commit:
Fix JavaDoc comments in DialogBox and MainWindow

Commit:
Fix JavaDoc comments in Megamind class

Commit:
Updated JavaDocs in Storage class to better reflect behaviour

Commit:
Update JavaDocs in Deadline class to better reflect behaviour

Commit:
Update JavaDocs in Event class to better reflect behaviour

Commit:
Update JavaDocs in List class to better reflect behaviour

Commit:
Update JavaDocs in Task class to better reflect behaviour

Commit:
Update JavaDocs in Todo class to better reflect behaviour

Commit:
Update Megamind logic to save TaskList after every command
